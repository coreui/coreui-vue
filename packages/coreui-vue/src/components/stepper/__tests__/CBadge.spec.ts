import { shallowMount } from '@vue/test-utils'
import { CBadge as Component } from '../../'

const ComponentName = 'CBadge'
const wrapper = shallowMount(Component)
const customWrapper = shallowMount(Component, {
  props: {
    color: 'success',
  },
  attrs: {
    class: 'bazinga',
  },
  slots: {
    default: 'Hello World!',
  },
})

describe(`Loads and display ${ComponentName} component`, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('renders correctly with slot', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})

describe(`Customize ${ComponentName} component`, () => {
  it('has a prope class names', () => {
    expect(customWrapper.classes('bazinga')).toBe(true)
    expect(customWrapper.classes('badge')).toBe(true)
    expect(customWrapper.classes('bg-success')).toBe(true)
  })
  it('default slot contains text', () => {
    expect(customWrapper.text()).toBe('Hello World!')
  })
})
