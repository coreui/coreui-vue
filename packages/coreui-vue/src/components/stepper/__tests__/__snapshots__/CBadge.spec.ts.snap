// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads and display CBadge component renders correctly 1`] = `
<span
  class="badge"
/>
`;

exports[`Loads and display CBadge component renders correctly with slot 1`] = `
<span
  class="badge bg-success bazinga"
>
  Hello World!
</span>
`;
